sdk/OoyalaSkinSDK/discoveryPanel.js

```diff
@@ -69,6 +81,30 @@ var DiscoveryPanel = React.createClass({
     }
   },
 
+  onStatusPressed: function() {
+    this.setState({showCircularStatus: false});
+  },
+
+  setCounterTime: function(time) {
+    this.setState({
+      currentCounterVal: time,
+      counterLimit: time,
+      showCircularStatus: true,
+    });
+  },
+
+  updateTimer: function(row) {
+    if (this.state.currentCounterVal == 0) {
+      this.onRowSelected(row);
+    } else {
+      var self = this;
+      setTimeout(function() {
```

Comment:
First, do you need to stop the timeout at any point, or does this go on forever?

I think its fine how it is currently.  I think i may be overcomplicating UI. @Matthew-C , do you have any opinions on this? using setTimeout in javascript for timers, and then using it to autoplay the next video?